docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: docs
+title: Best Practices
+permalink: /documentation/recipes/best-practices/best-practices.html
+---
+# Best Practices
+
+This article covers the recommended ways to test with TestCafe and covers the following topics:
+
+* Smart Assertions
+* Use of Page Objects
+* Use of Roles for Login
+* File structure
+* Clean up between Tests (beforeEach)
+* E2E Tests Limitations
+* Selectors strategy
+
+## Smart Assertions
+
+In end-to-end web testing, peripheral factors (like network lag, processor, or memory bottlenecks in containers) can interfere with the assertions and render tests inconclusive (otherwise known as 'flaky').
+
+Out of the box, TestCafe includes a [Smart Assertion Query Mechanism](../../guides/basic-guides/assert.md#smart-assertion-query-mechanism), which helps mitigate these effects. This mechanism introduces additional wait time for all the assertions: if an assertion fails, the test doesn't fail straight away. Instead, the assertion retries multiple times within a timeout. That helps to stabilize the tests without a big impact on performance.
+
+However, the smart assertions query mechanism only works with promises returned from TestCafe's own `Selector API`.
+
+The following example demonstrates the potential issue:
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with Selector', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    //this technique leads to unstable results. Avoid it.
```

Comment:
It's an **approach** rather than a technique, but since **the word** is pretty much banned...
Anyway:

>  //an awaited selector doesn't update and leads to unstable test results. Avoid it.